examples/AdvancedOperations/AddSmartDisplayAd.php

```diff
@@ -0,0 +1,436 @@
+        $campaignResponse = $campaignServiceClient->mutateCampaigns(
+            $customerId,
+            [$campaignOperation]
+        );
+
+        $campaignResourceName = $campaignResponse->getResults()[0]->getResourceName();
+        printf("Added a Smart Display campaign named '%s'.%s", $campaignResourceName, PHP_EOL);
+
+        return $campaignResourceName;
+    }
+
+    /**
+     * Creates an ad group.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the client customer ID
+     * @param string $campaignResourceName the resource name of the campaign
+     * @return string the resource name of the newly created ad group
+     */
+    private static function createAdGroup(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $campaignResourceName
+    ) {
+        // Constructs an ad group and set its type.
+        $adGroup = new AdGroup([
+            'name' => new StringValue(['value' => 'Earth to Mars Cruises #' . uniqid()]),
+            'campaign' => $campaignResourceName,
+            'status' => AdGroupStatus::PAUSED,
+        ]);
+
+        // Creates an ad group operation.
+        $adGroupOperation = new AdGroupOperation();
+        $adGroupOperation->setCreate($adGroup);
+
+        // Issues a mutate request to add the ad group.
+        $adGroupServiceClient = $googleAdsClient->getAdGroupServiceClient();
+        /** @var MutateAdGroupsResponse $adGroupResponse */
+        $adGroupResponse = $adGroupServiceClient->mutateAdGroups($customerId, [$adGroupOperation]);
+
+        // Print out some information about the added ad group.
+        $adGroupResourceName = $adGroupResponse->getResults()[0]->getResourceName();
+        printf("Added ad group named '%s'.%s", $adGroupResourceName, PHP_EOL);
+
+        return $adGroupResourceName;
+    }
+
+    /**
+     * Creates a responsive display ad, which is a recommended ad type for Smart Display campaigns.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the client customer ID
+     * @param string $adGroupResourceName the ad group resource name
+     * @param string|null $marketingImageAssetResourceName optional, the resource name of marketing
+     *     image asset
+     * @param string|null $squareMarketingImageAssetResourceName optional, the resource name of
+     *     square marketing image asset
+     */
+    private static function createResponsiveDisplayAd(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $adGroupResourceName,
+        string $marketingImageAssetResourceName = null,
+        string $squareMarketingImageAssetResourceName = null
+    ) {
+        // Creates a new image asset for marketing image and square marketing image if there are no
+        // assets' resource names specified.
+        $marketingImageAssetResourceName = $marketingImageAssetResourceName ?:
```

Comment:
Why not creating 'AdImageAsset' instances directly from here? This would reduce the size of the responsiveDisplayAdInfo variable construction statement.

Reply:
I believe you cannot do that. For images, you would need to use AssetService to create image assets first unlike AdTextAsset.

Reply:
Sorry I misunderstand your comment, but I think it's good as is. We need to create AdTextAsset, so this is more consistent and easier to read.

But it seems AdImageAsset instantiation can be shortened into one line. I'll do that soon.